src/core/src/core_logic/PatchInstaller.py

```diff
@@ -176,14 +177,49 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
         self.last_still_needed_packages = all_packages
         self.last_still_needed_package_versions = all_package_versions
 
+        stopwatch_for_batch_install_process = Stopwatch(self.env_layer, self.telemetry_writer, self.composite_logger)
+        stopwatch_for_batch_install_process.start()
+
+        installed_update_count, patch_installation_successful, maintenance_window_batch_cutoff_reached = self.install_packages_in_batches(all_packages, packages, package_versions, maintenance_window, package_manager)
+
+        stopwatch_for_batch_install_process.stop()
+
+        # Check for packages available and install them in batches. Some packages may not be installed due to:
+        # (a) Not enough time remaining in maintenance window
+        # (b) Failure during package installation
+        # (c) New packages added during the brief time batch installation was in progress
+        packages, package_versions = self.get_remaining_packages_to_install(package_manager)
+
+        batch_processing_perf_log = "[{0}={1}][{2}={3}][{4}={5}][{6}={7}][{8}={9}]".format(Constants.PerfLogTrackerParams.TASK, "InstallPackagesInBatches", "InstalledPackagesCountInBatchProcessing",
+                                    str(installed_update_count), "RemainingPackagesToInstall", str(len(packages)), Constants.PerfLogTrackerParams.PATCH_OPERATION_SUCCESSFUL, str(patch_installation_successful), 
+                                    "IsMaintenanceWindowBatchCutoffReached", str(maintenance_window_batch_cutoff_reached))
+
+        stopwatch_for_batch_install_process.write_telemetry_for_stopwatch(str(batch_processing_perf_log))
+
+        if len(packages) == 0:
+            installed_update_count = self.log_metrics_and_perform_final_reconciliation(package_manager, maintenance_window, patch_installation_successful, maintenance_window_exceeded, installed_update_count)
+            return installed_update_count, patch_installation_successful, maintenance_window_exceeded
+        else:
+            progress_status = self.progress_template.format(str(datetime.timedelta(minutes=maintenance_window.get_remaining_time_in_minutes())), str(self.attempted_parent_package_install_count), str(self.successful_parent_package_install_count), str(self.failed_parent_package_install_count), str(installed_update_count - self.successful_parent_package_install_count),
+                                                        "Following packages are not installed in batch installation: " + str(packages))
+            self.composite_logger.log(progress_status)
+
         for package, version in zip(packages, package_versions):
+            if package not in self.last_still_needed_packages:
```

Comment:
Add stopwatch for sequential patching so that time taken by sequential patching is captured.

Reply:
Done.